### ISSUE TITLE
Problem with \ in the generated mkdocs.yml when deploying pydoc-markdown content with Netlify

### ISSUE THREAD

**Issue by StarfallProjects** [NiklasRosenstein/pydoc-markdown#129]
Hi! I have a situation where everything is working locally, both with `mkdocs serve` and if I do `mkdocs build` then locally serve the results.

However, when I deploy to Netlify, I get 404s for all the pages generated by pydoc-markdown. A little digging showed that this is because a normal `mkdocs.yml` has the filepaths in the nav with `/` (e.g. `myDir/myfile.md`), but pydoc-markdown's autogenerated `mkdocs.yml` uses `\` (e.g. `myDir\myfile.md`) It seems local mkdocs can handle this, but Netlify can't - it generates the following warnings:

```
11:04:26 AM: INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
11:04:26 AM:   - pythonv1/cape_privacy.pandas/dtypes.md

11:04:26 AM: WARNING -  A relative path to 'pythonv1/cape_privacy.pandas\dtypes.md' is included in the 'nav' configuration, which is not found in the documentation files
```

Manually editing the autogenerated `mkdocs.yml` to us `/` fixed this.

I suspect this isn't entirely an issue with pydoc-markdown (I am using a combination of pydoc-markdown, Spotify's Monorepo plugin, and then building and deploying with Netlify). So I understand if you don't feel this is something you need to fix. However I thought I'd raise it as it does seem like the `mkdocs.yml` you generate may be non-standard, and possibly this issue will help future users.

**Versions:**
pydoc-markdown 3.1.1
python 3.7

**Comment by NiklasRosenstein** [comment]
I have no clue how Netlify works. Could it be that they are building the documentation on a Windows OS? That would explain the backslashes as the `Page.filename()` function uses `os.path.join()`, and the `MkDocsRenderer` use that filename when generating the Mkdocs config.

https://github.com/NiklasRosenstein/pydoc-markdown/blob/a615c26f3faeb045a6fdc0b90a19fe1b393ed0bf/pydoc-markdown/src/pydoc_markdown/util/pages.py#L48-L62

https://github.com/NiklasRosenstein/pydoc-markdown/blob/a615c26f3faeb045a6fdc0b90a19fe1b393ed0bf/pydoc-markdown/src/pydoc_markdown/contrib/renderers/mkdocs.py#L135-L136

https://github.com/NiklasRosenstein/pydoc-markdown/blob/a615c26f3faeb045a6fdc0b90a19fe1b393ed0bf/pydoc-markdown/src/pydoc_markdown/contrib/renderers/mkdocs.py#L105-L117

**Comment by StarfallProjects** [comment]
You've hit the nail on the head - I'm building on Windows (with a subsequent build step on Netlify). So the issue is my deviant choice of operating system.

**Comment by NiklasRosenstein** [comment]
Just to recap, you are producing the `mkdocs.yml` file via Pydoc-Markdown on a Windows machine and then send it for building to Netlify, which uses an OS with forward-slashes?

I'm wondering if that could be a common workflow in which case we probably want to make sure that the `mkdocs.yml` produced by Pydoc-Markdown is valid across platforms.

**Comment by StarfallProjects** [comment]
Yeah Netlify uses Linux (Ubuntu I think), and it's very popular for deploying static sites, including docs sites.

Our docs are going to be a bit of a beast, so there's a bash script to bring it all together - for now, this is my workaround:

```
# Generate the Cape Python docs
echo "Generating Cape Python docs . . . "
pydoc-markdown
# pydoc-markdown on Windows generates mkdocs.yml with '\', which Netlify cannot handle: 
# https://github.com/NiklasRosenstein/pydoc-markdown/issues/129
# using '#' as a delimeter to make this monstrosity less monstrous
sed -i -e 's#\\#\/#g' ./pythonv1/mkdocs.yml
```

**Comment by NiklasRosenstein** [comment]
Hey @StarfallProjects , I made it so the paths in the generated `mkdocs.yml` use forward slashes on Windows. Can you give it a try without your `sed` command and Pydoc-Markdown v.3.2.0?

**Comment by StarfallProjects** [comment]
Looks good! Thank you - I really didn't expect a fix for this, as it's a "Windows weirdness" issue really.

**Comment by guillaume-ion** [comment]
Hi, I don't mean to unearth an old thread however I have a very similar issue with the docusaurus sidebar file generation. 
I work on windows and the file is generated with "\\" instead of "/" is there an option to toggle somewhere ? 

Thank you.

**Example:**

```
{
  "items": [
    {
      "items": [
        "edison\\reference\\ion_sdk\\edison_api\\edison_api"
      ],
      "label": "ion_sdk.edison_api",
      "type": "category"
    },
    {
      "items": [
        "edison\\reference\\ion_sdk\\tools\\utils"
      ],
      "label": "ion_sdk.tools",
      "type": "category"
    }
  ],
  "label": "ion_sdk",
  "type": "category"
}
```

**Comment by NiklasRosenstein** [comment]
@guillaume-ion , thanks for raising the issue. Should be fixed in 3.10.2!
